src/com/facebook/buck/apple/AppleBundle.java

```diff
@@ -838,10 +898,26 @@ public CxxPreprocessorInput getCxxPreprocessorInput(
             cxxPlatform,
             headerVisibility);
       }
+      if (isFrameworkBundle() && headerVisibility == HeaderVisibility.PUBLIC) {
```

Comment:
I was thinking more like Java's interfaces with default methods, but it seems like they're only available from 8 and onwards.

The methods for implementing `CxxPreprocessorDep, NativeLinkable` as a famework bundle are very short, so I'll see if it makes sense to put them in an apple frameworks utility class.